### PR TITLE
 grid_sample 

### LINKED ISSUE #21957: grid_sample

### PR DESCRIPTION
**Pull request by JONAHKYAGABA**
Close #21957

### CODE REVIEW

File: ivy/functional/frontends/paddle/nn/functional/vision.py [hunk 1]
```diff
@@ -170,3 +170,105 @@ def pixel_shuffle(x, upscale_factor, data_format="NCHW"):
     return ivy.reshape(
         ivy.permute_dims(input_reshaped, (0, 1, 4, 2, 5, 3)), (b, oh, ow, oc)
     )
+@to_ivy_arrays_and_back
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/paddle/nn/functional/vision.py:173]: We need two lines of whitespace between functions

File: ivy/functional/frontends/paddle/nn/functional/vision.py [hunk 2]
```diff
@@ -170,3 +170,105 @@ def pixel_shuffle(x, upscale_factor, data_format="NCHW"):
...
+    else:
+        grid_x = 0.5 * grid_x * (W - 1)
+        grid_y = 0.5 * grid_y * (H - 1)
+
+    y_t = ivy.zeros((N, C, grid_H, grid_W), dtype=x.dtype)
+
+    for n in range(N):
+        for c in range(C):
+            for i in range(grid_H):
+                for j in range(grid_W):
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/paddle/nn/functional/vision.py:220]: It should be possible to do this without using nested loops, which would be far more efficient. Maybe have a look at this implementation of grid_sample which was recently implemented in the torch frontend for inspiration? https://github.com/unifyai/ivy/pull/22539

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_vision.py [hunk 3]
```diff
@@ -11,6 +11,8 @@
 # --------------- #
 
 
+
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_vision.py:14]: We need to add a paddle frontend test for this function.

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
Thanks for the contribution @JONAHKYAGABA! I've left some comments, please could you take a look? Thanks!

File: ivy/functional/frontends/paddle/nn/functional/vision.py (hunk 1 above)
Comment (JONAHKYAGABA) [reply ivy/functional/frontends/paddle/nn/functional/vision.py:175]: Thank you @Sam-Armstrong for the corrections here

File: ivy/functional/frontends/paddle/nn/functional/vision.py (hunk 2 above)
Comment (JONAHKYAGABA) [reply ivy/functional/frontends/paddle/nn/functional/vision.py]: Thank you @Sam-Armstrong for the resources provided have been able to remove the forloops now lert me proceed to the tests

**Pull request closed without merging**